pkg/exporters/k8sexporter/k8s_exporter.go

```diff
@@ -38,13 +39,23 @@ type k8sExporter struct {
 	conditionManager condition.ConditionManager
 }
 
-// NewExporterOrDie creates a exporter for Kubernetes apiserver exporting, panics if error occurs.
+// NewExporterOrDie creates a exporter for Kubernetes apiserver exporting,
+// panics if error occurs.
+//
+// Note that this function may be blocked (until a timeout occurs) before
+// kube-apiserver becomes ready.
 func NewExporterOrDie(npdo *options.NodeProblemDetectorOptions) types.Exporter {
 	if !npdo.EnableK8sExporter {
 		return nil
 	}
 
 	c := problemclient.NewClientOrDie(npdo)
+
+	glog.Infof("Waiting for kube-apiserver to be ready (timeout %v)...", npdo.APIServerWaitTimeout)
+	if err := waitForAPIServerReadyWithTimeout(c, npdo); err != nil {
+		glog.Warningf("kube-apiserver did not become ready: %v", err)
```

Comment:
What's the error message when timeout happens? Just want to make sure something like `time out waiting for the kube-apiserver` is included in the warning message.

Reply:
Added more info to the message.

For example:
```
W0709 10:30:01.060769  231463 k8s_exporter.go:56] kube-apiserver did not become ready: timed out on waiting for kube-apiserver to return the node object: timed out waiting for the condition
```